providers/domainnameshop/convert.go

```diff
@@ -0,0 +1,89 @@
+package domainnameshop
+
+import (
+	"strconv"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/miekg/dns/dnsutil"
+)
+
+func toRecordConfig(domain string, currentRecords *domainNameShopRecord) *models.RecordConfig {
```

Comment:
Suggestion: Rename `currentRecords` to `currentRecord` if it only stores one record.